jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -266,31 +252,29 @@ private static void addIncludePaths(@NotNull GeneralCommandLine commandLine, @Nu
   @NotNull
   private static List<String> getErlangModulePaths(@NotNull ErlangTarget target,
                                                    @NotNull CompileContext context,
-                                                   @NotNull List<String> dirtyFilePaths,
                                                    boolean isTest) {
-    List<ErlangFileDescriptor> fileDescriptors = getErlangFileDescriptorFromConfig(target, context, isTest);
-    return fileDescriptors != null ?
-           getSortedErlangModulePathsToCompile(fileDescriptors, dirtyFilePaths) :
-           getErlangModulePathsDefault(target, isTest);
+    List<String> erlangBuilderDirtyFilesInfo = getErlangModulePathsFromConfig(target, context, isTest);
+    return erlangBuilderDirtyFilesInfo != null ? erlangBuilderDirtyFilesInfo : getErlangModulePathsDefault(target, isTest);
   }
 
-  @NotNull
-  private static List<String> getSortedErlangModulePathsToCompile(@NotNull List<ErlangFileDescriptor> sortedModuleDescriptors,
-                                                                  @NotNull List<String> dirtyModules) {
-    SortedModuleDependencyGraph semiGraph = new SortedModuleDependencyGraph(sortedModuleDescriptors);
-    GraphGenerator<Node> graph = GraphGenerator.create(semiGraph);
-    markDirtyNodes(semiGraph.getNodesByName(dirtyModules), graph);
-    return ContainerUtil.mapNotNull(semiGraph.getNodes(), new Function<Node, String>() {
-      @Nullable
-      @Override
-      public String fun(Node node) {
-        return node.myDirty && FileUtilRt.extensionEquals(node.myErlangModulePath, "erl") ? node.myErlangModulePath : null;
-      }
-    });
+  @Nullable
+  private static List<String> getErlangModulePathsFromConfig(@NotNull ErlangTarget target,
+                                                             @NotNull CompileContext context,
+                                                             boolean isTests) {
+    String fileName = ErlangBuilderUtil.ORDERED_DIRTY_FILES_PREFIX + target.getModule().getName() + ".xml";
+    ErlangModuleBuildOrder erlangModuleBuildOrder = readFromXML(fileName, context, ErlangModuleBuildOrder.class);
+    if (erlangModuleBuildOrder == null) return null;
+
+    List<String> modules = erlangModuleBuildOrder.myOrderedErlangFilePaths;
+    if (isTests) {
+      return ContainerUtil.concat(modules, erlangModuleBuildOrder.myOrderedErlangTestFilePaths);
+    }
+    return modules;
   }
 
   @NotNull
   private static List<String> getErlangModulePathsDefault(@NotNull ErlangTarget target, boolean isTests) {
+    LOG.warn("Erlang module " + target.getModule().getName() + " will be fully rebuilt.");
```

Comment:
Yes, it should complain if we don't have a build order supplied to it. 

But I don't like we get these reports when running tests - it's distracting and you also need to figure out that it's ok it's fully rebuilt as we're running tests - remembering that in, say, half a year is not that easy )

I suggest we replace this log message with a build message reported to compile context, and assert this message is issued.